### PR TITLE
DX-1714 add webhook event handler docs

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link

<br>

https://deploy-preview-5586--tyk-docs.netlify.app/docs/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic/#set-up-a-webhook-event-handler-in-tyk-operator

https://deploy-preview-5586--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/reference/api-definition/

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

___

### **PR Type**
Documentation

___

### **Description**
- Added detailed documentation on how to set up a webhook event handler using Tyk Operator, including configuration examples.
- Updated the Tyk Operator API definition reference to reflect support for webhook event handlers, with a link to the relevant documentation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>event-webhook-tyk-classic.md</strong><dd><code>Add documentation for webhook event handler setup in Tyk Operator</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md

<li>Added documentation for setting up a webhook event handler in Tyk <br>Operator.<br> <li> Provided example configuration for <code>event_handlers</code> in ApiDefinition <br>CRD.<br> <li> Explained integration with Tyk Classic API Definition.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5586/files#diff-9d798ca0f42835c02158139b4a85c0fff6db8b711f7386d790a6c3089b2556fc">+34/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>api-definition.md</strong><dd><code>Update feature support for webhook event handler in Tyk Operator</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/reference/api-definition.md

<li>Updated feature support table to indicate webhook event handler <br>support.<br> <li> Added reference link to the webhook event handler documentation.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5586/files#diff-2a3c59b9948a8a992bebbaecc84925ccbe04564820a47e3274cd189823ddea14">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md [hunk 1]
```diff
@@ -84,3 +84,36 @@ Now:
...
+    target_url: http://httpbin.org
+    listen_path: /webhook-handler
+    strip_listen_path: true
+  event_handlers:
+    events:
+      AuthFailure:
+      - handler_name: "eh_web_hook_handler"
+        handler_meta:
+          method: "POST"
+          target_path: "http://posttestserver.com/post.php?dir=tyk-event-test"
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md]: let's also add webhook name here
```suggestion
        handler_meta:
          method: "POST"
          name: "webhook name"
          target_path: "http://posttestserver.com/post.php?dir=tyk-event-test"
```

**Review by buraksekili — COMMENTED** [review]
lgtm with a minor suggestion

**Review by buraksekili — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md [hunk 2]
```diff
@@ -84,3 +84,37 @@ Now:
 Note that you can register multiple webhooks to be triggered in response to a single event and you can register the same webhook with multiple API events.
 
 Remember to click **Save** to save your changes.
+
+## Set up a webhook event handler in Tyk Operator
+
+Tyk Operator supports event handler integration for Tyk Classic API Definition. By configuring the `event_handlers` field in ApiDefinition Custom Resource Definition (CRD), you can enable webhooks to be triggered by specific API events. 
+
+The process for configuring webhook event handlers using Tyk Operator is similar to that explained in [Set up a webhook event handler in the Tyk Classic API Definition](#set-up-a-webhook-event-handler-in-the-tyk-classic-api-definition). The example API Definition below enabled event handler by setting `spec.event_handlers`.
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md]:
```suggestion
The process for configuring webhook event handlers using Tyk Operator is similar to that explained in [Set up a webhook event handler in the Tyk Classic API Definition](#set-up-a-webhook-event-handler-in-the-tyk-classic-api-definition). The example API Definition below enables the event handler by setting `spec.event_handlers`.
```

File: tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md [hunk 3]
```diff
@@ -84,3 +84,37 @@ Now:
 Note that you can register multiple webhooks to be triggered in response to a single event and you can register the same webhook with multiple API events.
 
 Remember to click **Save** to save your changes.
+
+## Set up a webhook event handler in Tyk Operator
+
+Tyk Operator supports event handler integration for Tyk Classic API Definition. By configuring the `event_handlers` field in ApiDefinition Custom Resource Definition (CRD), you can enable webhooks to be triggered by specific API events.
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-classic.md]:
```suggestion
Tyk Operator supports event handler integration for Tyk Classic API Definition. Configuring the `event_handlers` field in ApiDefinition Custom Resource Definition (CRD) enables webhooks to be triggered by [specific API events]({{< ref "basic-config-and-security/report-monitor-trigger-events/event-types" >}}). 
```

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**